wire-ios-data-model/Source/Model/UserClient/UserClient.swift

```diff
@@ -246,18 +246,19 @@ public class UserClient: ZMManagedObject, UserClientType {
     }
 
     /// Resets releationships and ends an exisiting session before deleting the object
-    /// Call this from the syncMOC only
-    public func deleteClientAndEndSession() {
+    public func deleteClientAndEndSession() async {
+        try? await deleteSession()
```

Comment:
suggestion: what about logging an error here?

```suggestion
        do {
            try await deleteSession()
        } catch {
            WireLogger.userClient.error("error deleting session: \(String(reflecting: error))")
        }
```

Reply:
I'll add a log but the better solution is to let the function throw.